handler/datasets.go

```diff
@@ -0,0 +1,193 @@
+	if err != nil {
+		log.Error(ctx, "error parsing the metadata uri", err)
+		return err
+	}
+
+	// Mapping Json to Avro
+	versionDetails := models.VersionDetails{
+		ReleaseDate: datasetMetadataPublished.ReleaseDate,
+	}
+
+	datasetDetailsData := models.DatasetDetails{
+		Title:          datasetMetadataPublished.Title,
+		Summary:        datasetMetadataPublished.Description,
+		CanonicalTopic: datasetMetadataPublished.CanonicalTopic,
+		Subtopics:      datasetMetadataPublished.Subtopics,
+		Edition:        edition,
+		DatasetID:      datasetID,
+		Type:           "dataset_landing_page",
+		PopulationType: models.PopulationType{},
+		Dimensions:     []models.Dimension{},
+	}
+
+	if err := populateCantabularFields(ctx, datasetMetadataPublished, &datasetDetailsData); err != nil {
+		return fmt.Errorf("failed to populate cantabular fields: %w", err)
+	}
+
+	if datasetMetadataPublished.Keywords != nil {
+		datasetDetailsData.Keywords = *datasetMetadataPublished.Keywords
+	}
+
+	versionMetadata := models.CMDData{
+		UID:            generatedID,
+		URI:            parsedURI.Path,
+		VersionDetails: versionDetails,
+		DatasetDetails: datasetDetailsData,
+	}
+
+	datasetVersionMetadata := models.MapVersionMetadataToSearchDataImport(versionMetadata)
+	logData = log.Data{
+		"datasetVersionData": datasetVersionMetadata,
+	}
+	log.Info(ctx, "datasetVersionMetadata ", logData)
+
+	datasetVersionMetadata.TraceID = cpEvent.TraceID
+	datasetVersionMetadata.JobID = cpEvent.JobID
+	datasetVersionMetadata.SearchIndex = getIndexName(cpEvent.SearchIndex)
+	datasetVersionMetadata.DataType = "dataset_landing_page"
+
+	// Marshall Avro and sending message
+	if sdImportErr := h.Producer.SearchDataImport(ctx, datasetVersionMetadata); sdImportErr != nil {
+		log.Fatal(ctx, "error while attempting to send DatasetAPIImport event to producer", sdImportErr)
+		return sdImportErr
+	}
+	return nil
+}
+
+// populateCantabularFields sets the dimensions, type and population type for Cantabular datasets, according to the type in the provided metadata.
+// The provided DatasetDetails model will be mutated, adding type, dimensions and population type
+func populateCantabularFields(ctx context.Context, metadata dataset.Metadata, dd *models.DatasetDetails) error {
+	if dd == nil {
+		return errors.New("nil DatasetDetails provided")
+	}
+
+	if metadata.DatasetDetails.IsBasedOn == nil {
+		return nil // is_based_on not present in Dataset
+	}
+
+	t := metadata.DatasetDetails.IsBasedOn.Type
+	if _, isCantabular := CantabularTypes[t]; !isCantabular {
+		return nil // Dataset type is not Cantabular
+	}
+
+	dd.Type = t
+	log.Info(ctx, "identified dataset with cantabular type", log.Data{
+		"type":           t,
+		"num_dimensions": len(metadata.Dimensions)},
+	)
+
+	dd.Dimensions = []models.Dimension{}
+	for i := range metadata.Dimensions {
+		// Using pointers to prevent copying lots of data.
+		// TODO consider changing type to []*VersionDimension in dp-api-clients-go
```

Comment:
We probably need to wait for new client planned to be created inside the dataset API - nothing to do here and happy for you to leave this comment in here for now